File: app/src/main/java/io/github/sithengineer/motoqueiro/PermissionAuthority.java
package io.github.sithengineer.motoqueiro;

import rx.Observable;

public interface PermissionAuthority {
  // todo improve the permission request system to show a rationale when asking for permissions
  void askForPermissions(String[] permissionRequest, int requestCode);

  Observable<PermissionResponse> getPermissionResult(int requestCode);
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/PermissionAuthority.kt
package io.github.sithengineer.motoqueiro

import io.reactivex.Observable

interface PermissionAuthority {
  // todo improve the permission request system to show a rationale when asking for permissions
  fun askForPermissions(permissionRequests: Array<String>, requestCode: Int)

  fun getPermissionResult(requestCode: Int): Observable<PermissionResponse>
}
